bliss/models/galaxy_net.py

```diff
@@ -65,30 +67,42 @@ def __init__(
         super().__init__()
         self.save_hyperparameters()
 
-        self.main_encoder = CenteredGalaxyEncoder(slen=slen, latent_dim=latent_dim, n_bands=n_bands)
-        self.main_decoder = CenteredGalaxyDecoder(slen=slen, latent_dim=latent_dim, n_bands=n_bands)
+        self.main_encoder = CenteredGalaxyEncoder(
+            slen=slen,
+            latent_dim=(latent_dim // 2) * 2,
+            n_bands=n_bands,
+        )
+        self.main_decoder = CenteredGalaxyDecoder(
+            slen=slen, latent_dim=latent_dim // 2, n_bands=n_bands
+        )
         self.main_autoencoder = nn.Sequential(self.main_encoder, self.main_decoder)
 
         self.residual_encoder = CenteredGalaxyEncoder(
-            slen=slen, latent_dim=latent_dim, n_bands=n_bands
+            slen=slen,
+            latent_dim=(latent_dim // 2) * 2,
+            n_bands=n_bands,
         )
         self.residual_decoder = CenteredGalaxyDecoder(
-            slen=slen, latent_dim=latent_dim, n_bands=n_bands
+            slen=slen, latent_dim=latent_dim // 2, n_bands=n_bands
         )
         self.residual_autoencoder = nn.Sequential(self.residual_encoder, self.residual_decoder)
 
+        self.register_buffer("prior_mean", torch.tensor(0.0))
+        self.register_buffer("prior_var", torch.tensor(1.0))
+        self.dist_main = Normal(self.prior_mean, self.prior_var)
+        self.dist_residual = Normal(self.prior_mean, self.prior_var)
```

Comment:
does this defeat the purpose of `ensure_dist_on_device` or viceversa?

Reply:
For some reason these registered buffers weren't getting properly moved to the device when the prior is inside a data loader...

Reply:
It's ugly but I couldn't find a better way to get it to work

Reply:
I see, yeah I suspected something like that... It's a weird issue